Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionMapsView.cs

```diff
@@ -57,43 +62,49 @@ public ActionMapsView(VisualElement root, StateContainer stateContainer)
             m_ListView.RegisterCallback<ValidateCommandEvent>(OnValidateCommand);
 
             CreateSelector(s => new ViewStateCollection<string>(Selectors.GetActionMapNames(s)),
-                (actionMapNames, state) => new ViewState(Selectors.GetSelectedActionMap(state), actionMapNames));
+                (actionMapNames, state) => new ViewState(
+                    Selectors.GetSelectedActionMap(state), actionMapNames?.ToList() ?? new List<string>()));
 
             m_AddActionMapButton = root.Q<Button>("add-new-action-map-button");
             m_AddActionMapButton.clicked += AddActionMap;
             ContextMenu.GetContextMenuForActionMapListView(this, m_ListView.parent);
         }
 
-        public override void RedrawUI(ViewState viewState)
+        protected override void RedrawUI(ViewState viewState)
         {
-            m_ListView.itemsSource = viewState.actionMapNames?.ToList() ?? new List<string>();
-            if (viewState.selectedActionMap.HasValue)
+            m_ListView.itemsSource = viewState.actionMapNames;
+            m_ListView.Rebuild();
+
+            // Update view to reflect model selection
+            var desiredSelectedIndex = viewState.selectedActionMapIndex;
+            if (desiredSelectedIndex >= 0 && desiredSelectedIndex != m_ListView.selectedIndex)
             {
-                var indexOf = viewState.actionMapNames.IndexOf(viewState.selectedActionMap.Value.name);
-                m_ListView.SetSelection(indexOf);
+                m_ListView.SetSelection(desiredSelectedIndex);
+                m_ListView.ScrollToItem(desiredSelectedIndex);
             }
-            m_ListView.Rebuild();
-            RenameNewActionMaps();
         }
 
         public override void DestroyView()
         {
             m_AddActionMapButton.clicked -= AddActionMap;
         }
 
-        private void RenameNewActionMaps()
+        private void RenameNewActionMap()
         {
-            if (!m_EnterRenamingMode)
-                return;
-            m_ListView.ScrollToItem(m_ListView.selectedIndex);
             var element = m_ListView.GetRootElementForIndex(m_ListView.selectedIndex);
+            m_ListView.ScrollToItem(m_ListView.selectedIndex);
```

Comment:
Why are we scrolling to the item twice? Isn't this triggered from RedrawUI already after adding the new Action Map?

Reply:
Thanks for pointing this out, agree it seems to be done twice. However I still think its possible to select an item in the list, then scroll so the selected item isn't visible, then hit Rename shortcut. If the item being renamed is not visible then it might get confusing?

Reply:
right, I didn't consider this case, anyway I think it won't harm at least.